tools/functional-tester/etcd-agent/agent.go

```diff
@@ -93,9 +94,18 @@ func (a *Agent) stop() error {
 
 // restart restarts the stopped etcd process.
 func (a *Agent) restart() error {
-	a.cmd = exec.Command(a.cmd.Path, a.cmd.Args[1:]...)
-	a.cmd.Stdout = a.logfile
-	a.cmd.Stderr = a.logfile
+	flags := a.cmd.Args[1:]
+	for i := range flags {
```

Comment:
it does not really matter. restarting an etcd cluster with datadir will ignore all `initial` flags.